packages/bitcore-wallet-service/src/lib/server.ts

```diff
@@ -589,6 +598,9 @@ export class WalletService {
       // only for testing
       if (opts.doNotMigrate) return cb(null, wallet);
 
+      // backwards compatibility
+      if (!wallet.chain) wallet.chain = ChainService.getChain(wallet.coin);
```

Comment:
TODO: we should probably create a migration script to add & populate the `chain` field for every wallet in the db

FYI, some very near future updates that are required for polygon will have breaking changes and require a major version bump. Introducing a breaking change and having a migration script to run pre-upgrade would be appropriate here.
Having said that, what you have here works.